### ISSUE TITLE
Rogue under Cloak and Paladin's Judgement and Hammer of Wrath

### ISSUE THREAD

**Issue by Renehadox** [gamefreedomgit/Maelstrom#2784]
[//]: # (REMBEMBER! Add links to things related to the bug using for example:)
[//]: # (http://wowhead.com/)
[//]: # (cata-twinhead.twinstar.cz)

**Description:**

When Rogue is under Cloak of Shadows and a Paladin uses Judgement or Hammer of Wrath it should go through it since Seal of Truth doesn't apply magic debuff.

Currently, when a rogue uses cloak here on this server and a paladin casts Judgement or HoW it says " resisted " 

**Database links:**

We can see that in the video below:

https://www.youtube.com/watch?v=mlsqFWgwTXc 

06:44 Ret uses judgement while rogue is under cloak and it applies the debuff.

**Comment by Renehadox** [comment]
Also I noticed something yesterday, when rogue uses cloak and a ret paladin uses crusader strike or templar's on him it says immune, which is also bugged, it shouldn't say immune, since those are not magical attacks.

**Comment by Renehadox** [comment]
![WoWScrnShot_040623_011706](https://user-images.githubusercontent.com/67479994/230243380-0a08a164-9e9c-4fdd-9c0a-13edbb277298.jpg)
![WoWScrnShot_040623_011619](https://user-images.githubusercontent.com/67479994/230243391-585b7bce-79f0-47b4-a761-8d8b29274a5a.jpg)

I just did a test duel, it says immune, also targets are immune in BG ???

**Comment by UltraNix** [comment]
Tested again and still Hammer of Wrath goes through Cloak of Shadow.
![asdf](https://user-images.githubusercontent.com/80540499/230357439-d1c9aece-2222-4086-af10-8c48c5097e18.png)

What are rogue talents?

**Comment by Renehadox** [comment]
> 

He is sub.

**Comment by Renehadox** [comment]
> Tested again and still Hammer of Wrath goes through Cloak of Shadow. ![asdf](https://user-images.githubusercontent.com/80540499/230357439-d1c9aece-2222-4086-af10-8c48c5097e18.png)
> 
> What are rogue talents?

He is sub.

**Comment by UltraNix** [comment]
Show me talent tree.

**Comment by Renehadox** [comment]
How can i show you talent tree?

Guy is offline at the moment, whats your point?

What talents have to do with Hammer missing targets???

**Comment by UltraNix** [comment]
Becasue I wanna know if there is something else that makes rogue immune to Hammer.

**Comment by Renehadox** [comment]
> Becasue I wanna know if there is something else that makes rogue immune to Hammer.

No there is not, there is no such talent that makes rogue immune to hammer.

**Comment by UltraNix** [comment]
As I can see, Cloak of Shadows gives 1 sec of total immunity to all magical spells after use (school immunity).
Both, Hammer and Judgement are holy schools and that's why it's immuned.
In that movie in 1st post, paladin uses Judgement while rogue has Cloak of Shadow at least for 3 sec already.

**Comment by UltraNix** [comment]
Also, I found this https://www.mmo-champion.com/threads/794727-Cloak-of-Shadow?p=9517463&viewfull=1#post9517463

**Comment by Renehadox** [comment]
Hi, where did you see that?

https://wotlk-twinhead.twinstar.cz/?spell=31224

**Comment by Renehadox** [comment]
> As I can see, Cloak of Shadows gives 1 sec of total immunity to all magical spells after use (school immunity).
> Both, Hammer and Judgement are holy schools and that's why it's immuned.
> In that movie in 1st post, paladin uses Judgement while rogue has Cloak of Shadow at least for 3 sec already.

This,

" As I can see, Cloak of Shadows gives 1 sec of total immunity to all magical spells after use (school immunity).
Both, Hammer and Judgement are holy schools and that's why it's immuned.
In that movie in 1st post, paladin uses Judgement while rogue has Cloak of Shadow at least for 3 sec already. "

Where did you find this?

**Comment by UltraNix** [comment]
From DBC (aka Blizzard) files.
When you use Cloak of Shadows, all harmfull effects are removed and spell `81549` is casted.
```
ID - 81549 Cloak of Shadows
=================================================

Category = 0, SpellIconID = 1933, activeIconID = 0, SpellVisual = (0,0)
Family SPELLFAMILY_ROGUE, flag [0] 0x40000000 [1] 0x00000000 [2] 0x00000000

SpellSchoolMask = 1 (SPELL_SCHOOL_MASK_NORMAL)
DamageClass = 0 (SPELL_DAMAGE_CLASS_NONE)
PreventionType = 0 (SPELL_PREVENTION_TYPE_NONE)
=================================================
Attributes: 0x00040190 (SPELL_ATTR0_ABILITY, SPELL_ATTR0_HIDDEN_CLIENTSIDE, SPELL_ATTR0_HIDE_IN_COMBAT_LOG, SPELL_ATTR0_DONT_AFFECT_SHEATH_STATE)
AttributesEx1: 0x00000420 (SPELL_ATTR1_NOT_BREAK_STEALTH, SPELL_ATTR1_NO_THREAT)
=================================================
Spell Level = 66, base 66, max 0, maxTarget 0

Category = 0
DispelType = 0 (DISPEL_NONE)
Mechanic = 0 (MECHANIC_NONE)
SpellRange: (Id 1) "Self Only":
    MinRangeNegative = 0, MinRangePositive = 0
    MaxRangeNegative = 0, MaxRangePositive = 0

Cast time (Id 1) = 0,00
Duration: ID (36)  1000, 0, 1000
Interrupt Flags: 0x00000000, AuraIF 0x00000000, ChannelIF 0x00000000
Chance = 0, charges - 0
=================================================
Effect 0: Id 6 (SPELL_EFFECT_APPLY_AURA)
BasePoints = 0
Targets (1, 0) (TARGET_UNIT_CASTER, NO_TARGET)
Aura Id 39 (SPELL_AURA_SCHOOL_IMMUNITY), value = 0, misc = 126 (126), miscB = 0, periodic = 0
```
This spell grants you schools immunities for all spell schools but physical (`misc = 126`) for 1 sec. At that time, you are immuned to all non-physical spells.
Hammer of Wrath and Judgement are both Holy spells, so rogue is immuned to these two.

Cloak of Shadows also gives you this aura:
```
Effect 0: Id 6 (SPELL_EFFECT_APPLY_AURA)
BasePoints = -200
Targets (1, 0) (TARGET_UNIT_CASTER, NO_TARGET)
Aura Id 186 (SPELL_AURA_MOD_ATTACKER_SPELL_HIT_CHANCE), value = -200, misc = 126 (126), miscB = 0, periodic = 0
```
That aura makes you to lower attacker's hit on magic spells to 0.
Hammer of Wrath and Judgement  are both melee spells, that;s why that aura does not affect it. But still, you can resist/miss with these two, but CoS has nothing to do with it.

TL;DR: I don;t see any bug here.

**Comment by Renehadox** [comment]
> Hammer of Wrath and Judgement are both melee spells:

TL;DR: i read the whole thing even tho i don't have any knowledge about developing.

However, you are mistaken in one point.

Hammer of Wrath and Judgement are both RANGED spells and not melee.

There is a difference, right? ;)

**Comment by UltraNix** [comment]
AS you can see, they are marked as melee spells above in DBC files. Though - in many cases they are both considered as ranged spells. 
But still - it has no impact on Cloak of Shadows - only magical spells (non-melee, non-ranged) are affected.
Closing as not bug.

**Comment by Renehadox** [comment]
> AS you can see, they are marked as melee spells above in DBC files. Though - in many cases they are both considered as ranged spells. But still - it has no impact on Cloak of Shadows - only magical spells (non-melee, non-ranged) are affected. Closing as not bug.

Do not close it yet.

On all of the cata servers i played, ( Old AT, Molten-Wow, Monster-Wow ) i mentioned those as a more stabled ones.

Cloak of Shadows worked like I reported.

Hammer of Wrath, Judgement, Crusader and Templar should pierce through Cloak.

It also worked like that on Retail Cataclysm, which I personally played.

**Comment by UltraNix** [comment]
And it does, like I showed you in many screenshots above. Even on your SS:
https://user-images.githubusercontent.com/67479994/230243380-0a08a164-9e9c-4fdd-9c0a-13edbb277298.jpg
Judgement goes through Cloak.
Other private servers does not interesting me.

**Comment by Renehadox** [comment]
> 

I just mentioned them as an example cause they used be very popular Cata servers back in teh days ( fixed ones ).

Yes it does, but why at first it says immune?

Like you said it should work like that for 1 second for all attacks or what?

**Comment by Misxfits** [comment]
I  just logging to post a bug about this spell too, its not only that Pala spell, is tons of spells. Just go to bg click on ""Cloak of Shadows" and see the debuff zone grow up.
